modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotPartitionsVerifyHandler.java

```diff
@@ -106,8 +108,23 @@ public SnapshotPartitionsVerifyHandler(GridCacheSharedContext<?, ?> cctx) {
 
         SnapshotMetadata meta = opCtx.metadata();
 
-        Set<Integer> grps = F.isEmpty(opCtx.groups()) ? new HashSet<>(meta.partitions().keySet()) :
-            opCtx.groups().stream().map(CU::cacheId).collect(Collectors.toSet());
+        Set<Integer> grps = F.isEmpty(opCtx.groups())
+            ? new HashSet<>(meta.partitions().keySet())
+            : opCtx.groups().stream().map(CU::cacheId).collect(Collectors.toSet());
+
+        grps = grps.stream().filter(grp -> {
+            if (grp == MetaStorage.METASTORAGE_CACHE_ID)
+                return true;
+
+            CacheGroupDescriptor grpDesc = cctx.kernalContext().cache().cacheGroupDescriptor(grp);
+
+            if (grpDesc == null)
```

Comment:
I don't like this semantic. It is not so obvious that we restore, not create. The quick verify handler and the normal partition verify handlers work is different phases: CREATE and RESORE. We could use it. 

Suggestion:
`if (type() == SnapshotHandlerType.CREATE) {
    grps = grps.stream().filter(grp -> {
        if (grp == MetaStorage.METASTORAGE_CACHE_ID)
            return true;

        IgnitePredicate<ClusterNode> filter = cctx.kernalContext().cache().cacheGroupDescriptor(grp).config()
            .getNodeFilter();

        return filter == null || CU.affinityNode(cctx.localNode(), filter);
    }).collect(Collectors.toSet());
}`